### DISCUSSION TITLE
Use PyBaMM battery models for battery pack simulation

### DISCUSSION THREAD

**Discussion by wigging** [pybamm-team/PyBaMM#1616] [Ideas]
Are there any examples of using the battery models in PyBaMM to simulate an entire battery pack? If not, are there any future plans to include such a feature?

**Comment by brosaplanella** [comment]
Hi Gavin! There is no in-built capability yet, but I think @TomTranter has been doing some work on it and might be able to tell you more.

**Comment by TomTranter** [comment]
Hi @wigging yes this is something I've been working on. Going to publish a paper on it soon and deciding whether to release the code here or possibly develop it further and offer it as something commercial. What are you interested in using it for? Academia or Industry?

**Reply by wigging** [reply]
I'm interested in using it for a research project to study the effects of load profiles on battery packs. The project just got started and I'm currently evaluating what open source tools are available for modeling battery packs. This seems to be an active area in the battery community so I would prefer to avoid developing duplicate code by collaborating with others. There seems to be plenty of open source battery cell codes available (PyBaMM for example) but I haven't found anything for battery pack modeling.

**Reply by TomTranter** [reply]
I am not aware of any others either. LTSpice is free and can be used to model batteries but not using physics based models. I would be happy to share what I have done in the future if you are interested in collaborating. Drop me an email at t.tranter@ucl.ac.uk if you like and I'll let you know more details

**Reply by wigging** [reply]
I sent you an email to continue this discussion.

**Reply by wigging** [reply]
I received an automated email reply that you are on leave and will return August 9th. You might want to fix the date or remove the response if you're not on leave.
